src/Lykke.Job.TxDetector/Handlers/TransactionHandler.cs

```diff
@@ -0,0 +1,138 @@
+using Common;
+using Common.Log;
+using Lykke.Cqrs;
+using Lykke.Job.TxDetector.Commands;
+using Lykke.Job.TxDetector.Core;
+using Lykke.Job.TxDetector.Core.Domain.BitCoin;
+using Lykke.Job.TxDetector.Core.Domain.BitCoin.Ninja;
+using Lykke.Job.TxDetector.Core.Domain.Settings;
+using Lykke.Job.TxDetector.Core.Services.BitCoin;
+using Lykke.Job.TxDetector.Events;
+using Lykke.Job.TxDetector.Models;
+using Lykke.Job.TxDetector.Sagas;
+using Lykke.Job.TxDetector.Utils;
+using Lykke.Service.Assets.Client.Custom;
+
+namespace Lykke.Job.TxDetector.Handlers
+{
+    public class TransactionHandler
+    {
+        private readonly AppSettings.TxDetectorSettings _settings;
+        private readonly ISrvBlockchainReader _srvBlockchainReader;
+        private readonly IBalanceChangeTransactionsRepository _balanceChangeTransactionsRepository;
+        private readonly IInternalOperationsRepository _internalOperationsRepository;
+        private readonly ICachedAssetsService _assetsService;
+        private readonly IConfirmedTransactionsRepository _confirmedTransactionsRepository;
+        private readonly IPostponedCashInRepository _postponedCashInRepository;
+        private readonly IAppGlobalSettingsRepositry _appGlobalSettingsRepositry;
+
+        private readonly ILog _log;
+
+        public TransactionHandler(
+            IBalanceChangeTransactionsRepository balanceChangeTransactionsRepository,
+            IInternalOperationsRepository internalOperationsRepository,
+            ICachedAssetsService assetsService,
+            IConfirmedTransactionsRepository confirmedTransactionsRepository,
+            IPostponedCashInRepository postponedCashInRepository,
+            IAppGlobalSettingsRepositry appGlobalSettingsRepositry,
+            ILog log,
+            AppSettings.TxDetectorSettings settings,
+            ISrvBlockchainReader srvBlockchainReader)
+        {
+            _balanceChangeTransactionsRepository = balanceChangeTransactionsRepository;
+            _internalOperationsRepository = internalOperationsRepository;
+            _assetsService = assetsService;
+            _confirmedTransactionsRepository = confirmedTransactionsRepository;
+            _postponedCashInRepository = postponedCashInRepository;
+            _appGlobalSettingsRepositry = appGlobalSettingsRepositry;
+            _log = log;
+            _settings = settings;
+            _srvBlockchainReader = srvBlockchainReader;
+        }
+
+        // entry point
+        public async Task Handle(ProcessTransactionCommand command, IEventPublisher eventPublisher)
+        {
+            await _log.WriteInfoAsync(nameof(TransactionHandler), nameof(ProcessTransactionCommand), command.ToJson(), "");
+
+            var confirmations = await _srvBlockchainReader.GetConfirmationsCount(command.TransactionHash);
+            var isConfirmed = confirmations >= _settings.TxDetectorConfirmationsLimit;
+            if (!isConfirmed)
+            {
+                throw new Exception();
```

Comment:
This exception will not produce any spam in logs, right? Is it just a retry for current step to wait for a confirmations?

Reply:
It will spam. I'll change the implementation.